bson/src/main/org/bson/codecs/configuration/ChildCodecRegistry.java

```diff
@@ -19,14 +19,19 @@
 
 import org.bson.codecs.Codec;
 
-// An implementation of CodecRegistry that is used to detect cyclic dependencies between Codecs
-class ChildCodecRegistry<T> implements CodecRegistry {
+/**
+ * An implementation of CodecRegistry that is used to detect cyclic dependencies between Codecs
+ *
+ * @param <T> the {@code Codec} type
+ * @since 3.12
+ */
+public class ChildCodecRegistry<T> implements CodecRegistry {
```

Comment:
Making this public so that custom CodecRegistry implementations (like the one added for UuidRepresentation overriding) can implement cycle detection without too much trouble.

Reply:
Change my mind about this.  After moving the new codec registry into the bson module this could remain internal